substratools/workspace.py

```diff
@@ -147,41 +94,33 @@ def __init__(
                 makedir_safe(d)
 
 
-class CompositeAlgoWorkspace(OpenerWorkspace):
+class MetricsWorkspace(OpenerWorkspace):
+    """Filesystem workspace for metrics execution."""
+
     def __init__(
         self,
+        opener_path,
         dirpath=None,
         input_data_folder_paths=None,
         input_predictions_path=None,
-        input_head_model_path=None,
-        input_trunk_model_path=None,
-        output_head_model_path=None,
-        output_trunk_model_path=None,
-        output_predictions_path=None,
+        output_perf_path=None,
         log_path=None,
         chainkeys_path=None,
     ):
         super().__init__(
             dirpath=dirpath,
             input_data_folder_paths=input_data_folder_paths,
-            input_predictions_path=input_predictions_path,
-            output_predictions_path=output_predictions_path,
         )
 
-        self.input_head_model_path = input_head_model_path
-        self.input_trunk_model_path = input_trunk_model_path
-
-        self.output_head_model_path = output_head_model_path
-        self.output_trunk_model_path = output_trunk_model_path
-
+        self.output_perf_path = output_perf_path or self._get_default_path(DEFAULT_OUTPUT_PERF_PATH)
         self.log_path = log_path or self._get_default_path(DEFAULT_LOG_PATH)
-
         self.chainkeys_path = chainkeys_path or self._get_default_path(DEFAULT_CHAINKEYS_PATH)
+        self.input_predictions_path = input_predictions_path or self._get_default_path(DEFAULT_INPUT_PREDICTIONS_PATH)
+        self.opener_path = opener_path
 
-        dirs = [
-            self.chainkeys_path,
```

Comment:
Why isnt it needed anymore ? If so you I guess you can remove self.chainkeys_path, It seems it's not used elsewhere

Reply:
good catch
do not remove the chainkey path, it is part of the chainky support and we should not change/delete this feature
(`self.chainkeys_path` is exposed to the user, they use it to get the chainkey file)

Reply:
We should test this somewhere so we are sure there is no regression on it

Reply:
Sorry and good catch thank you :)